Sources/NIOConcurrencyHelpers/lock.swift

```diff
@@ -132,11 +148,17 @@ public final class ConditionLock<T: Equatable> {
     /// - Parameter value: The initial value to give the state variable.
     public init(value: T) {
         self._value = value
-        self.mutex = Lock()
 #if os(Windows)
+InitializeSRWLock(self.mutex)
         InitializeConditionVariable(self.cond)
 #else
-        let err = pthread_cond_init(self.cond, nil)
+        var attr = pthread_mutexattr_t()
+        pthread_mutexattr_init(&attr)
+        pthread_mutexattr_settype(&attr, .init(PTHREAD_MUTEX_ERRORCHECK))
+
+        var err = pthread_mutex_init(self.mutex, &attr)
+        precondition(err == 0, "\(#function) failed in pthread_mutex with error \(err)")
```

Comment:
It's a bit of a shame to repeat this lock initialisation code. Can we pull it out into a `fileprivate` top-level function so we can just call it in two places?